src/zenml/logging/step_logging.py

```diff
@@ -269,6 +270,9 @@ def __init__(
         # Immutable filesystems state
         self.last_merge_time = time.time()
 
+        # Lock for shared state
+        self.lock = threading.Lock()
```

Comment:
I think just to be save we should use an `RLock` here

Reply:
Makes sense, switched it.